Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -36,6 +40,25 @@ public async Task<IEnumerable<ScheduleDTO>> GetByUserPrimarySkillAsync(Guid proj
             return _mapper.Map<List<ScheduleDTO>>(schedules);
         }
 
+        public async Task BulkAppointOrCancelInterviewsAsync(IEnumerable<InterviewAppointmentDTO> interviewAppointmentDTOs, Guid projectId)
+        {
+            var candidates = await _candidateService.GetCandidatesByIdsAsync(interviewAppointmentDTOs.Select(c => c.CandidateId));
+            var scheduleCandidateInfos = GetScheduleCandidateInfos(candidates, interviewAppointmentDTOs, projectId);
+            var appointInterviewSlots = interviewAppointmentDTOs.Select(a => new InterviewAppointmentSlot()
+            {
+                ScheduleSlot = new ScheduleSlot()
+                {
+                    AvailableTime = a.AppointmentDateTime,
+                    ScheduleCandidateInfo = _mapper.Map<ScheduleCandidateInfo>(scheduleCandidateInfos.FirstOrDefault(c => c.Id == a.CandidateId)),
```

Comment:
You return shedule candidates info's not for all candidate , so in one moment you'll have mapping from null. Don't do it, check that candidate exists in your info's and map only after

Reply:
if scheduleCandidateInfos don't have infos by candidateId, i will get null and ScheduleCandidateInfo will be null